### ISSUE TITLE
com.android.phone high cpu usage and battery drain

### ISSUE THREAD

**Issue by harpowned** [GrapheneOS/os-issue-tracker#2239]
Phone: Pixel 6a - bluejay
GrapheneOS version: TQ2A.230505.002.2023060700
The issue also manifested on the previous version.
Carrier: Simyo Spain.
The carreir uses the Orange Spain network, so the IMSI starts with 21403

The phone feels warm to the touch at all times, and the battery lasts about 12h with light usage(which is around 1/3 of the usual battery life before the issue started).

I connected via adb shell, and ran the top command:
![imatge](https://github.com/GrapheneOS/os-issue-tracker/assets/6563361/b1eb4736-1c54-41ac-9366-82f24a249b9b)

As you can see, the com.android.phone app is constantly using a lot of CPU. It oscillates between 50 and 90%, but never below 50.
The phone has an uptime of 15 hours at the time of this screenshot, and this app reports having used almost 160 hours of cpu time.

I also did the following tests (while looking at top through adb shell):
* Airplane mode: CPU usage is Normal
* Wifi on, cellular off: CPU usage is Normal
* Cellular on, wifi off, cellular data off: CPU usage is High
* Cellular on, wifi off, cellular data on: CPU usage is High
* Cellular on, wifi on, cellular data on: CPU usage is High

In short, the problem manifests itself whenever the cellular connection is enabled, it's not related to the cellular data being active, or to an IP network connection.

Looking at adb log, I see a cyclic pattern, which looks like it's constantly reconnecting to the network somehow, but I'm not an Android developer, and this goes beyond my expertise. I

I'm attaching one minute of adb log for analysis.
[bluejay_1minute.log](https://github.com/GrapheneOS/os-issue-tracker/files/11706636/bluejay_1minute.log)

**Comment by muhomorr** [comment]
> The issue also manifested on the previous version.

Was the previous version the first version where this issue started occurring?

**Comment by harpowned** [comment]
> Was the previous version the first version where this issue started occurring?

I think so, yes, but I'm not 100% sure. I was busy with other stuff and it took me a few days to notice I was charging the phone every day and not every other day.
Any more information I can give you, or tests I can run?

**Comment by muhomorr** [comment]
A major update (Android 13 QPR3) is expected today or tomorrow, which is likely to address this issue.

**Comment by harpowned** [comment]
Ok, I'll wait for the update, and report back so we can hopefully close the ticket.
Thanks for the reply.
